### DISCUSSION TITLE
Memory leak?

### DISCUSSION THREAD

**Discussion by SparkJiao** [Jingkang50/OpenPSG#46] [Q&A] [answered]
I think there may be memory leakage.

The memory usage is increasing with the training process and has cost more than 100GB before the first epoch ends.

Did anyone encounter similar problem?

**Comment by Jingkang50** [comment, marked as answer]
Have you solved it yet?

**Reply by SparkJiao** [reply]
Yes, I use `del` command to free the memory used by inputs after each forward. At least it works for me.

**Reply by Jingkang50** [reply]
Cool, that pretty makes sense.

**Reply by cxzhang93** [reply]
> Yes, I use `del` command to free the memory used by inputs after each forward. At least it works for me.

Hi @SparkJiao , I was encountering same problem as you. could you elaborate how did you resolve this issue like what command you added to which location? Many thanks!
